Index: app/src/main/java/com/example/passfort/MainActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/passfort/MainActivity.kt b/app/src/main/java/com/example/passfort/root/MainActivity.kt
rename from app/src/main/java/com/example/passfort/MainActivity.kt
rename to app/src/main/java/com/example/passfort/root/MainActivity.kt
--- a/app/src/main/java/com/example/passfort/MainActivity.kt	(revision c50d75f593cd78072bf814a7e210316a15ac9bd9)
+++ b/app/src/main/java/com/example/passfort/root/MainActivity.kt	(date 1742939376716)
@@ -1,6 +1,8 @@
-package com.example.passfort
+package com.example.passfort.root
 
+import NavigationGraph
 import android.os.Bundle
+import android.util.Log
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
@@ -17,9 +19,17 @@
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        /*fb.collection("users").document()
+            .set(mapOf("name" to "Konda","email" to "konda@gmail.com"))*/
+
+
         enableEdgeToEdge()
         setContent {
-            PassFortTheme {
+            val navController = rememberNavController()
+            NavigationGraph(navController = navController)
+
+            /*PassFortTheme {
                 Scaffold(
                     modifier = Modifier
                         .fillMaxSize(),
@@ -30,7 +40,7 @@
                 ) { innerPadding ->
                     Greeting(
                         name = "Android",
-                        modifier = Modifier.padding(innerPadding)
+                        modifier = Modifier.padding(innerPadding)*/
                     )
                 }
             }
